src/main/java/com/google/research/bleth/simulator/AbstractSimulation.java

```diff
@@ -68,15 +74,21 @@ void updateObserversAwaknessState() {
      */
     void beaconsToObservers() {
         for (Beacon beacon : beacons) {
+            boolean observed = false; // for stats
             Transmission transmission = beacon.transmit();
             for (Observer observer : observers) {
                 if (observer.isAwake()) {
                     double distance = distance(beacon.getLocation(), observer.getLocation());
                     if (distance <= transmissionThresholdRadius) {
                         observer.observe(transmission);
+                        observed = true; // for stats
                     }
                 }
             }
+            if (observed) {
+                beaconsObservedPercent.put(String.valueOf(beacon.getId()),
+                        beaconsObservedPercent.getOrDefault(String.valueOf(beacon.getId()), 0D) + 1); // for stats
```

Comment:
`beaconsObservedPercent` actually holds the observations count, not percent. Consider changing its name or alternatively divide by number of rounds each iteration.